Sparkle/SUBasicUpdateDriver.m

```diff
@@ -420,17 +418,30 @@ - (void)installWithToolAndRelaunch:(BOOL)relaunch displayingUserInterface:(BOOL)
     }
 
     NSBundle *sparkleBundle = [NSBundle bundleWithIdentifier:SUBundleIdentifier];
-
     // Copy the relauncher into a temporary directory so we can get to it after the new version's installed.
     // Only the paranoid survive: if there's already a stray copy of relaunch there, we would have problems.
     NSString *const relaunchPathToCopy = [sparkleBundle pathForResource:[[sparkleBundle infoDictionary] objectForKey:SURelaunchToolNameKey] ofType:@"app"];
     if (relaunchPathToCopy != nil) {
         NSString *targetPath = [self.host.appCachePath stringByAppendingPathComponent:[relaunchPathToCopy lastPathComponent]];
-        // Only the paranoid survive: if there's already a stray copy of relaunch there, we would have problems.
+        
+        NSFileManager *fileManager = [[NSFileManager alloc] init];
+        
+        BOOL shouldAbortUpdate = NO;
```

Comment:
Instead of using a state variable, could you move the code to a function, so `if (!shouldAbortUpdate &&` becomes `if (![self doSomeStuff] && [`?

This code is fine, but I'm afraid such variables can cause trouble later if somebody moves the code around.